shared/controlflow/codeql/controlflow/Cfg.qll

```diff
@@ -0,0 +1,1293 @@
+/**
+ * Provides a shared interface and implementation for constructing control-flow graphs
+ * (CFGs) from abstract syntax trees (ASTs).
+ */
+
+private import codeql.util.Location
+
+/** Provides the language-specific input specification. */
+signature module InputSig<LocationSig Location> {
+  /** The base class for `ControlFlowTree`. */
+  class ControlFlowTreeBase {
+    /** Gets a textual representation of this element. */
+    string toString();
+
+    /** Gets the location of this element. */
+    Location getLocation();
+  }
+
+  /** An AST node. */
+  class AstNode extends ControlFlowTreeBase;
```

Comment:
Do we really need `AstNode` and `ControlFlowTreeBase` in the signature?

Reply:
No, you are right, I have merged them.